pkg/stanza/tokenize/multiline.go

```diff
@@ -107,34 +109,53 @@ func LineStartSplitFunc(re *regexp.Regexp, flushAtEOF bool, trimFunc trim.Func)
 		}
 
 		if firstMatchEnd == len(data) {
+			if omitPatternFromRecord {
+				// read more data and try again
+				return firstMatchEnd, nil, nil
+			}
+
 			// the first match goes to the end of the bufer, so don't look for a second match
 			return 0, nil, nil
 		}
 
 		// Flush if no more data is expected
 		if atEOF && flushAtEOF {
-			token = trimFunc(data)
+			if omitPatternFromRecord {
+				token = trimFunc(data[firstMatchEnd:])
+			} else {
+				token = trimFunc(data)
+			}
 			advance = len(data)
 			return
 		}
 
 		secondLocOfset := firstMatchEnd + 1
 		secondLoc := re.FindIndex(data[secondLocOfset:])
 		if secondLoc == nil {
-			return 0, nil, nil // read more data and try again
+			if omitPatternFromRecord {
+				// read more data and try again
+				return firstMatchEnd, nil, nil
+			}
```

Comment:
as I remember it didn't work without it. the condition is saying that secondLoc is null which means that this is the last time we see the pattern. If I'm not returning " firstMatchEnd, nil, nil " then since there are no more matches for this pattern we stop scanning and the last token is from 0 .. stringEnd (which is actually the current string left to scan)

Reply:
OK, I ran the tests without it and it seemed to pass, but when I did manual tests it didn't.
maybe I misunderstood something, but for this test
{
	Name:        "OneLogSimpleOmitPattern",
	Pattern:     `LOGSTART`,
	Input:       []byte("LOGSTART 123 log1 LOGSTART 123 a"),
	ExpectedTokens: []string{
		`LOGSTART 123 log1`,
	},
},
		
why the expected output isn't:
ExpectedTokens: []string{
				`LOGSTART 123 log1`,
				`LOGSTART 123 a`,
			},

Reply:
> OK, I ran the tests without it and it seemed to pass, but when I did manual tests it didn't. maybe I misunderstood something, but for this test
> 
> ```
> {
> 	Name:        "OneLogSimpleOmitPattern",
> 	Pattern:     `LOGSTART`,
> 	Input:       []byte("LOGSTART 123 log1 LOGSTART 123 a"),
> 	ExpectedTokens: []string{
> 		`LOGSTART 123 log1`,
> 	},
> },
> ```
> 
> why the expected output isn't:
> 
> ```
> ExpectedTokens: []string{
> 				`LOGSTART 123 log1`,
> 				`LOGSTART 123 a`,
> 			},
> ```

The idea is that we don't know whether or not the second line is complete until we see another line start pattern. We don't make an assumption that each log is written to the file atomically, so we do not consider that the second line is complete just because we reach the end of the file. (This is one reason the `force_flush_period` setting exists.)

A similar example may be easier to visualize. Consider if we see the following in a file:

```
LOGSTART We know this log is complete because we can see the next start pattern.
LOGSTART However, this log is not complete becau
```